src/Core/Components/Dialog/FluentDialog.razor

```diff
@@ -13,6 +13,7 @@
                    aria-describedby=@(_parameters.AriaDescribedby ?? AriaDescribedby)
                    aria-labelledby=@(_parameters.AriaLabelledby ?? AriaLabelledby)
                    aria-label=@(_parameters.AriaLabel ?? AriaLabel)
+                   @ondialogdismiss=HandleDismissAsync
```

Comment:
Is this change necessary for this PR spcifically?

Reply:
The original intention behind the change was because dismissing a dialog doesn't go down the same path as when it is closed by clicking the "X" or by clicking the Secondary action. By putting the dismiss under a similar path, checking for the "OnDialogClosing" callback would only need to exist in one place.

After looking into this further, any `@ondialogdismiss` added to `<FluentDialog />` is overwrites the one listed here because of the `@attributes=@AdditionalAttributes` bind. I will move it back in a follow up commit.

Reply:
I've addressed this in 41e1c1cd69a9993652e15e23241bcdfe08fb534d